### PR TITLE
oob, ood, and error is now handled with pop up, also refinements to f…

### PR DESCRIPTION
**Pull request by eutychius**
…unction 'traverseAnchorLinks'

### CODE REVIEW

File: app/src/main/java/at/tbd/offlinebrowser/MainActivity.java [hunk 1]
```diff
@@ -98,7 +85,13 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
             @Override
             public boolean shouldOverrideUrlLoading(WebView view, String url)//Deprecated, but other needs min API 21..
             {
```
Comment (kroney) [inline app/src/main/java/at/tbd/offlinebrowser/MainActivity.java]: Please reformat

File: app/src/main/java/at/tbd/offlinebrowser/WebsiteDownloader.java [hunk 2]
```diff
@@ -119,7 +122,7 @@ private String RecursiveSiteDownload(String url, int currentDepth) {
         fileName = SaveHtmlStringToFile(doc.toString(), fileName);
 
         if (fileName == null) {
-            fileName = "Error saving file";
+            fileName = "404:"+url;//own error format?
```
Comment (kroney) [inline app/src/main/java/at/tbd/offlinebrowser/WebsiteDownloader.java]: No, own format needed. Probably won't matter to the user, why it did not work to download
Please remove the comment

File: app/src/main/java/at/tbd/offlinebrowser/WebsiteDownloader.java [hunk 3]
```diff
@@ -131,32 +134,43 @@ private void TraverseImages(Document doc) {
         for (Element el : pics) {
             if (mExistingPics.containsKey(el.attr("src"))) {
                 el.attr("src", mExistingPics.get(el.attr("src")));
+            }else {
```
Comment (kroney) [inline app/src/main/java/at/tbd/offlinebrowser/WebsiteDownloader.java]: empty else?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -28,6 +28,10 @@
     <string name="pref_oob_error_display_name">Always open error</string>
     <string name="pref_oob_domain_description">Never ask before opening not downloaded URL out of domain</string>
     <string name="pref_oob_error_description">Never ask before opening not downloaded URL due to error</string>
+    <string name="ood_error">Url is out of depth and was not downloaded, continue online?</string>
```
Comment (kroney) [inline app/src/main/res/values/strings.xml]: You'll have to merge this file with the master

**Comment by eutychius** [comment]
it is done

**Pull request merged**